### PR TITLE
Add Logging service using Sentry captureConsole

### PR DESCRIPTION
**Pull request by UncleBu1728**
`captureConcole` integration from Sentry and a customized function is added to log user information 
- console.error and console.info is now monitored by Sentry for logging purposes
- The implementation on` login.tsx `page is to monitor if the authing tokens are still valid when users login. Normally if a user logout should clean the local storage which should not contain keys of `_authing_token` nor `_authing_user`. Therefore, if the keys mentioned above exist when the user is directed to the login page, it could mean the authentication token is expired or invalid. The `logToSentry` function should send logging messages if such situations happen. 

con: captureConsole integration could capture any console logger errors from localhosts, but it could be filtered or removed by the searching options on Sentry/issue page

### CODE REVIEW

File: src/shared/logToSentry.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+// other details like timestamp, and hostname/ IP address should be included by Sentry
+const zLogObject = z.object({
+    message: z.string(),
+    email: z.string().email().optional(),
+    data: z.object({}).optional()
+})
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:7]: Add semi-colon. Also, in a different PR, can you help enable eslint check for semi-colons?

File: src/shared/logToSentry.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+// other details like timestamp, and hostname/ IP address should be included by Sentry
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:2]: add empty line

File: src/shared/logToSentry.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const zLogObject = z.object({
+    message: z.string(),
+    email: z.string().email().optional(),
+    data: z.object({}).optional()
+})
+
+enum logLevel{
+    ERROR,
+    INFO
+}
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:12]: semi-colon

File: src/shared/logToSentry.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+const logToSentry = (tag: keyof typeof logLevel, logDetail: logObject) => {
+    if ((tag) === 'ERROR') {
+        console.error('LOG', tag, JSON.stringify(logDetail));
+    }
+
+    if ((tag) === 'INFO') {
+        console.info('LOG', tag, JSON.stringify(logDetail));
+    }
+}
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:24]: semi-colon

File: src/shared/logToSentry.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+// other details like timestamp, and hostname/ IP address should be included by Sentry
+const zLogObject = z.object({
+    message: z.string(),
+    email: z.string().email().optional(),
+    data: z.object({}).optional()
+})
+
+enum logLevel{
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:9]: `LogLevel`

File: src/shared/logToSentry.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    ERROR,
+    INFO
+}
+
+export type logObject = z.TypeOf<typeof zLogObject>;
+
+const logToSentry = (tag: keyof typeof logLevel, logDetail: logObject) => {
+    if ((tag) === 'ERROR') {
+        console.error('LOG', tag, JSON.stringify(logDetail));
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:18]: I don't understand why you need the logToSentry function. Can't we just use console.log instead whereever logToSentry is called?

File: src/shared/logToSentry.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+// other details like timestamp, and hostname/ IP address should be included by Sentry
+const zLogObject = z.object({
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:3]: No need to use zod. define typescript types directly:

```
type LogObject = {
   ...
}
```

File: src/shared/logToSentry.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+})
+
+enum logLevel{
+    ERROR,
+    INFO
+}
+
+export type logObject = z.TypeOf<typeof zLogObject>;
+
+const logToSentry = (tag: keyof typeof logLevel, logDetail: logObject) => {
```
Comment (weihanwang) [inline src/shared/logToSentry.ts:16]: Why do you need the `LogObject` type? Why don't  just pass message, email, data as three function parameters?

File: src/pages/callback.tsx
Comment (weihanwang) [inline src/pages/callback.tsx:1]: Don't mix unrelated changes in one PR for reviews.

File: src/shared/logToSentry.ts (hunk 8 above)
Comment (UncleBu1728) [reply src/shared/logToSentry.ts:16]: I was thinking of creating this object type just in case if we need more keys or properties to be passed to log without changing the number of parameters of `logToSentry`, but I did a quick research about the inefficiency of JSON.stringfy and will change it to a list parameters

File: src/shared/logToSentry.ts (hunk 6 above)
Comment (UncleBu1728) [reply src/shared/logToSentry.ts:18]: In addition to what I have sent on WeChat, I was also thinking of keeping the logs in a consistent format and easier to manage

File: src/pages/login.tsx [hunk 9]
```diff
@@ -34,10 +34,10 @@ export default function Login() {
...
         <Container as="footer">
           <Link isExternal href="https://vercel.com/?utm_source=yuanjian&utm_campaign=oss">
-            <Image 
-            src={vercelBanner} 
-            alt="Vercel Banner"  
-            height="30"
+            <Image
+              src={vercelBanner}
+              alt="Vercel Banner"
+              height="30"
```
Comment (weihanwang) [inline src/pages/login.tsx:40]: can you revert this file? I'm removing it entirely.

File: sentry.client.config.ts [hunk 10]
```diff
@@ -3,19 +3,23 @@
...
 import * as Sentry from "@sentry/nextjs";
+import { CaptureConsole } from '@sentry/integrations';
 
 Sentry.init({
   dsn: "https://a1dc8cd72a62aacab0dd520eee207398@o4505687617175552.ingest.sentry.io/4505687622090752",
 
-  integrations: [new Sentry.BrowserTracing()],
+  integrations: [
+    new Sentry.BrowserTracing(),
+    new CaptureConsole({ levels: ['error', 'info'] }) // using error and info as customized tag logging service
```
[the remark this answers is not in the document]
Comment (weihanwang) [reply sentry.client.config.ts:13]: This is just a hypothesis, right? did you observe excessive messages already?

[the remark this answers is not in the document]
Comment (UncleBu1728) [reply sentry.client.config.ts:13]: The reason I changed the environment to 'testing' earlier was that I observed there were logger messages sent to Sentry even with the `ignoreErrors` from 'localhost'. So I was thinking of using the logger levels that are not used very often to log messages like "info" or "error". If you go to the sentry/issue page and check the console logger message in development, you will see that they are sent from localhost

**Review by weihanwang — APPROVED** [review]

**Review by weihanwang — APPROVED** [review]
